### PR TITLE
Run CMock unit tests on each pull request with a github action.

### PR DESCRIPTION
**Pull request by paulbartell**
Run CMock unit tests on each pull request with a github action.

Description
-----------
This PR adds a github workflow that runs the kernel CMock unit tests located in the FreeRTOS/FreeRTOS repository on each push to FreeRTOS / FreeRTOS-Kernel.

This workflow also uploads test coverage data to codecov.io so that it can be easily reviewed. Optionally, codecov.io can add a comment to the PR outlining the changes in coverage before and after the changes in the PR.

Test Steps
-----------
N/A

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        fetch-depth: 1
+    - name: Clone This Repo
+      uses: actions/checkout@v2
+      with:
+        path: ./FreeRTOS/Source
+
+    - name: Setup Python
+      uses: actions/setup-python@master
+      with:
+        python-version: 3.8
```
Comment (lundinc2) [inline .github/workflows/unit-tests.yml:23]: This is a vulnerable python verison. Please use 3.7.10 or a patched python version

Comment (paulbartell) [reply .github/workflows/unit-tests.yml:23]: "3.8" maps to the latest 3.8.X version, which is 3.8.8 and includes the fix for CVE-2021-3177.
Example run: https://github.com/paulbartell/FreeRTOS-Kernel/runs/2125390265?check_suite_focus=true
Release notes for 3.8.8: https://www.python.org/downloads/release/python-388/

**Comment by paulbartell** [comment]
Here's a link to an example coverage report on codecov.io:
https://codecov.io/github/paulbartell/FreeRTOS-Kernel/commit/186d213c24095f0bb9045dcd7bdb9692afa3aca7

**Review by lundinc2 — APPROVED** [review]

**Review by nateglims — DISMISSED** [review]

**Comment by paulbartell** [comment]
Waiting on this ticket to be completed to enable the codecov.io app: https://t.corp.amazon.com/V333420686

**Review by cobusve — APPROVED** [review]

**Pull request merged**